nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -252,6 +258,16 @@
             .defaultValue("false")
             .build();
 
+    static final PropertyDescriptor MAP_RECORD_TO_JSON = new PropertyDescriptor.Builder()
+            .name("put-db-record-map-record-to-json")
+            .displayName("Map \"record\" types to JSON")
```

Comment:
What if we made this a dropdown with allowable values vs booleans? It appears that setting the JDBC type to `OTHER` supports both JSON and JSONB (at least for PostgreSQL), but in the future we may want to support the SQLXML type for inserting into XML columns and so on. The description of the `JSON` option should probably explicitly say that the JDBC type is set to `OTHER`, as that might be helpful information for other databases that do things differently.

Reply:
The JSON support is implemented using the MapRecord API that converts the Record to JSON without having to use a record writer. Since nothing like that exists for XML, how do you want to handle it? I suppose we could allow a writer to be specified and have the writer simply take the schema from the Record field.

Reply:
I don't see where MapRecord converts Record to JSON, I thought you were using Jackson's ObjectMapper for that? For XML I'm leery of JAXB as I guess that has Java 9+ and/or classpath issues, but I wonder if XStream is an option. Also I don't know if external systems take YAML these days but the idea was to offer the capability to support various structured data formats. If everything but JSON is a non-starter then I'm fine with leaving it a boolean.

Reply:
What I meant was, I use `toMap` to convert to a vanilla Java `Map` and then use Jackson for that. Come to think of it, I probably could support XML that way...

Reply:
Going to dig into this and see if there's a way to detect what the real type of the field is and apply this automagically (json vs xml as appropriate):

https://github.com/FasterXML/jackson-dataformat-xml